searx/plugins/tor_check.py

```diff
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""
+A plugin to check if the ip address of the request is a TOR exit node
+if the user searches for "tor" "tor check" or "check tor"
+It fetches the tor exit node list from https://check.torproject.org/exit-addresses
+and parses all the IPs into a list, then checks if the user's ip address
+is in it.
+
+Enable in 'settings.yml' (uncomment "Tor check plugin"
+in the 'enabled_plugins' section)
+"""
+
+import re
+from urllib.request import urlopen
+from urllib.error import HTTPError
+
+name = "Tor check plugin"
+description = """
+This plugin checks if the address of the request is a TOR exit node,
+and informs the user if it is, like check.torproject.org but from searxng.
+"""
+default_on = False
+
+# Regex for exit node addresses in the list.
+reg = re.compile(r"(?<=ExitAddress )\S+")
+
+
+def post_search(request, search):
+
+    if search.search_query.query.lower() in ("tor", "tor check", "check tor"):
+
+        # Request the list of tor exit nodes.
+        try:
+            with urlopen("https://check.torproject.org/exit-addresses") as ex_nodes:
+                node_list = re.findall(reg, str(ex_nodes.read()))
+
+        except HTTPError:
+            # No answer, return error
+            search.result_container.answers["tor"] = {
+                "answer": "Tor exit node list (https://check.torproject.org/exit-addresses) unreachable."
+            }
+            return True
+
+        x_forwarded_for = request.headers.getlist("X-Forwarded-For")
+
+        if x_forwarded_for:
+            ip_address = x_forwarded_for[0]
+        else:
+            ip_address = request.remote_addr
+
+        if ip_address in node_list:
```

Comment:
Maybe it should be reworded. Regardless "Tor" should be capitalized.

Reply:
You're right. I'm going to improve that

Reply:
Sorry I didn't highlight the correct lines. Either I forgot how this works or my browser is acting up.